### PR TITLE
add support for twitter and refactor shared code

### PR DESCRIPTION
**Pull request by k-yle**
Twitter is the easiest platform so far, since it's text-based. So we don't need to extract text from images.

---

## Todo

- [x] support bluring tweets when viewing a users profile. To test: visit https://twitter.com/pizzahut
- [x] support bluring a specific tweet that is open in the popup. To test: visit https://twitter.com/pizzahut/status/1571159392451452939
- [x] refactor popup.js to de-duplicate code related to switches, since there are now many switches
- [x] moved shared code into a new file called  `shared.js`
- [x] moved scripts that are injected into the page into a new folder called `worker`, because there are now many files. 

## Motivation

See issue #51. The aim is to support multiple social media platforms

## Attached GitHub issue

Partially addresses #51.

### Documentation

- [x] ~~Updated the readme documents (`README.md` etc.)~~
- [x] New functions and methods have additional comments added to document their usage

### Local Build

- [x] Ran all pre-commit hooks `pre-commit run --all-files`
- [x] Extension has been tested to build correctly `cd extension && yarn && yarn build`
- [x] Backend has been tested to build correctly `cd backend && docker-compose build`
- [x] Extension test suite has been run locally `cd extension && yarn && yarn test`
- [x] Backend test suite has been run locally `cd backend && python3 -m unittest`

### GitHub

- [x] Target branch has been set correctly
- [x] PR has been rebased onto target branch
- [x] PR has been assigned an owner
- [x] Author has performed a self-review of the code
- [x] Removed the WIP message from the top of this PR

### CODE REVIEW

File: extension/public/worker/twitter.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/** @param {HTMLDivElement} el */
+function checkTweet(el) {
+  if (window.matchesBlocklist(el.innerText)) {
+    // eslint-disable-next-line no-param-reassign
```
Comment (Destiny-02) [inline extension/public/worker/twitter.js]: What does this comment do?

File: extension/public/worker/twitter.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    el.style.filter = "blur(2px)";
```
Comment (Destiny-02) [inline extension/public/worker/twitter.js]: Are you able to make the blur slightly stronger? I can kind of make out what the tweet says here. 

image

File: extension/public/popup.js [hunk 3]
```diff
@@ -26,28 +23,36 @@ const globalSwitchHandler = () => {
...
-  redditSwitch.checked = data.redditToggled;
-  redditSwitchHandler();
-});
-
-chrome.storage.sync.get(["globalToggled"], (data) => {
-  globalSwitch.checked = data.globalToggled;
-  globalSwitchHandler();
-});
+for (const storageKey in switches) {
+  const switchEl = switches[storageKey];
```
Comment (Destiny-02) [inline extension/public/popup.js]: It wasn't immediately obvious to me that `switchEl` = switch elements. Perhaps rename the variable. If this is a convention you've seen in other projects, you can ignore this comment.

**Review by Destiny-02 — CHANGES_REQUESTED** [review]
Thanks for the refactor - it is a much-needed change change as more platforms are being supported! The extension will be much more complete now that it works for Reddit, Instagram and Twitter. Please see comments for some questions/suggestions.

File: extension/public/worker/twitter.js (hunk 1 above)
Comment (k-yle) [reply extension/public/worker/twitter.js]: sorry, I've removed this comment. It silences a lint warnings from eslint, a bit like `@SuppressWarnings` in java. I've changed the config to disable this warning altogether, and I think we could even remove eslint once sonarlint/sonarcloud is properly configured (#75)

File: extension/public/popup.js (hunk 3 above)
Comment (k-yle) [reply extension/public/popup.js]: good point, this is a bad variable name. I've changed it to be more descriptive

**Review by Destiny-02 — APPROVED** [review]

**Pull request merged**